### PR TITLE
fix(p1-face): sync form control state

### LINKED ISSUE #771: [io-checkbox-group] Missing @Watch('required') — runtime required changes do not re-sync children
## Problem
`io-checkbox-group` has `@Watch('name')`, `@Watch('disabled')`, and `@Watch('error')` — all of which call `syncChildren()`. However, there is no `@Watch('required')`. Changing `required` at runtime (e.g., dynamically enabling required validation based on another field) does not trigger `syncChildren`, so the asterisk indicator in the legend updates but no downstream effect occurs.

## Current Behavior
Watchers in `io-checkbox-group.tsx` (lines 88–101):
```ts
@Watch('name')   onNameChange()     { this.syncChildren(); }
@Watch('disabled') onDisabledChange() { this.syncChildren(); }
@Watch('error')  onErrorChange()    { this.syncChildren(); }
// @Watch('required') is ABSENT
```

## Expected Behavior
Add `@Watch('required')` that calls `syncChildren()`, matching the pattern of all other watched props.

## Why This Matters
In dynamic forms where required status changes based on user selections, the missing watcher means required validation is silently stale. Framework integrations that bind `required` as a reactive prop are affected.

## Acceptance Criteria
- [ ] `@Watch('required')` is added and calls `syncChildren()`
- [ ] Unit test verifies that toggling `required` at runtime re-syncs children

## Test Requirements
- [ ] Unit test: set `required=false`, call `onRequiredChange()`, assert children have `required=false`; then set `required=true` and assert children have `required=true`

## Accessibility Requirements
N/A — this is a functional correctness issue that indirectly affects WCAG 1.3.1 via constraint validation.

### LINKED ISSUE #778: [io-radio-group] Missing formStateRestoreCallback — browser autofill/history state restoration not supported
## Problem
`io-radio-group` implements `formAssociated: true`, `formResetCallback`, and `formDisabledCallback`, but is missing `formStateRestoreCallback`. Without this lifecycle method, the browser cannot restore the group's selected value from autofill, session history, or page refresh when the form participates in the FACE API.

`io-checkbox` (the analogous FACE-enabled component) correctly implements all three callbacks including `formStateRestoreCallback`. `io-radio-group` only implements two.

## Current Behavior
`io-radio-group.tsx` has:
- `formResetCallback` (line 113)
- `formDisabledCallback` (line 120)
- `formStateRestoreCallback` — **missing**

## Expected Behavior
Add:
```ts
formStateRestoreCallback(state: string | null): void {
  this.value = state ?? '';
  this.syncChildren();
  this.updateTabStops();
  this.syncFormValue();
}
```

## Why This Matters
Browser autofill and back/forward cache rely on `formStateRestoreCallback` to rehydrate form state. Without it, returning to a page with a partially completed form will leave `io-radio-group` blank even when the browser attempts to restore state.

## Acceptance Criteria
- [ ] `formStateRestoreCallback(state: string | null)` is implemented
- [ ] It sets `this.value`, re-syncs children, updates tab stops, and calls `syncFormValue`
- [ ] Tests cover the callback

## Test Requirements
- [ ] Unit test: `formStateRestoreCallback('phone')` sets `value='phone'` and syncs children
- [ ] Unit test: `formStateRestoreCallback(null)` sets `value=''`

## Accessibility Requirements
N/A — functional completeness. Indirectly supports WCAG 3.3.4 by enabling pre-filled form state.

### LINKED ISSUE #794: [io-checkbox] FACE spec does not test indeterminate+required edge case — indeterminate should not count as "checked" for form validity
## Problem
`io-checkbox.syncFormValue()` evaluates validity as `required && !checked`. When `indeterminate=true` and `checked=false` (the typical initial state for a partially-selected parent checkbox), FACE correctly sets `valueMissing`. However, there is no test confirming this behavior, and there is also no test for the case where `indeterminate=true` and `required=true` — an edge case where a consumer might expect the indeterminate state to satisfy or not satisfy validity.

## Current Behavior
`syncFormValue` only checks `this.required && !this.checked`. There is no logic or test for how `indeterminate` interacts with `required` validity. The spec file covers `required+unchecked` and `required+checked` but not `required+indeterminate`.

## Expected Behavior
The behavior should be explicit and tested: `indeterminate=true` with `checked=false` and `required=true` should result in `valueMissing=true` (indeterminate does not constitute a form value). This should be documented in code comments and verified by a test.

## Why This Matters
Developers building "Select all" checkboxes that are `required` need predictable validity behavior. Undocumented edge cases lead to runtime surprises during form submission.

## Acceptance Criteria
- [ ] A comment in `syncFormValue` explicitly states indeterminate behavior with respect to validity
- [ ] Test added to `io-checkbox.face.spec.ts` covering `required=true` + `indeterminate=true` + `checked=false`

## Test Requirements
- [ ] Unit test: `required=true`, `indeterminate=true`, `checked=false` → `setValidity` called with `{ valueMissing: true }`
- [ ] Unit test: `required=true`, `indeterminate=true`, `checked=false` → `faceInvalid=true`

## Accessibility Requirements
N/A — test coverage gap. Correct behavior already exists; the gap is missing verification.

### LINKED ISSUE #804: [io-checkbox-group] syncChildren does not propagate required prop to child io-checkbox elements
## Problem
`io-checkbox-group.syncChildren()` propagates `name`, `disabled`, and `state` to slotted `io-checkbox` children, but it does not propagate `required`. As a result, each child checkbox has `required=false` even when the group declares `required=true`. The asterisk indicator appears in the legend, but the native `required` attribute is never set on any input, so constraint validation and FACE `valueMissing` are never triggered from the group.

## Current Behavior
`syncChildren` (line 149–158):
```ts
for (const checkbox of checkboxes) {
  checkbox.name = this.name;
  checkbox.disabled = this.disabled;
  checkbox.state = this.error ? 'error' : 'none';
  // required is NOT propagated
}
```

## Expected Behavior
When `io-checkbox-group` has `required=true`, at least one child `io-checkbox` should carry `required=true` so the browser and FACE can enforce selection. Additionally, a `@Watch('required')` should call `syncChildren` when the prop changes at runtime.

## Why This Matters
A form containing `<io-checkbox-group required>` will pass `form.checkValidity()` even when no checkbox is checked, silently submitting an empty selection. This is a silent form-submission correctness bug.

## Acceptance Criteria
- [ ] `syncChildren` propagates `required` to the first (or all) child `io-checkbox` elements, consistent with the semantics expected by the group
- [ ] A `@Watch('required')` decorator triggers `syncChildren` on prop change
- [ ] Existing unit test for `syncChildren` is extended to cover required propagation

## Test Requirements
- [ ] Unit test: group with `required=true` — first/all child checkboxes have `required=true` after `syncChildren`
- [ ] Unit test: changing `required` at runtime triggers `syncChildren`

## Accessibility Requirements
WCAG 1.3.1 Info and Relationships (Level A) — required state must be programmatically determinable.

### PR DESCRIPTION
**Pull request by jakeortega-io**
## Summary

- **#771** — `io-checkbox-group`: add `@Watch('required')` so runtime `required` changes re-sync children
- **#804** — `io-checkbox-group`: propagate `required` to child `io-checkbox` elements in `syncChildren`; extend lifecycle spec with required watcher and propagation tests
- **#778** — `io-radio-group`: add `formStateRestoreCallback` for browser autofill and history-state restore; add FACE spec coverage
- **#794** — `io-checkbox`: document that `indeterminate + required = valueMissing`; add 4 FACE spec edge cases for `indeterminate + required` combinations

## Scope

FACE / form-state behavior and tests only. No styling, token, or branding changes. Independent from PRs #885–#889.

## Test plan

- [ ] `npm run test` passes (4551 component + 1622 storefront — all green)
- [ ] `npm run governance:check` clean
- [ ] No forbidden terms in diff
- [ ] No CSS / styling changes

Fixes #771
Fixes #778
Fixes #794
Fixes #804

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: io-components/src/components/io-checkbox-group/io-checkbox-group.tsx [hunk 1]
```diff
@@ -148,11 +153,12 @@ export class IoCheckboxGroup {
...
     const checkboxes = Array.from(
-      this.el.querySelectorAll<HTMLElement & { name: string; disabled: boolean; value: string; state: string }>('io-checkbox'),
+      this.el.querySelectorAll<HTMLElement & { name: string; disabled: boolean; required: boolean; value: string; state: string }>('io-checkbox'),
     );
     for (const checkbox of checkboxes) {
       checkbox.name = this.name;
       checkbox.disabled = this.disabled;
+      checkbox.required = this.required;
       checkbox.state = this.error ? 'error' : 'none';
     }
```
[the remark this answers is not in the document]
Comment (jakeortega-io) [reply io-components/src/components/io-checkbox-group/io-checkbox-group.tsx:163]: Fixed in commit dab3303e. Removed propagation of 'required' to individual child checkboxes in syncChildren(). Group-level required validation is handled at the fieldset level, not by marking individual checkboxes as required.

File: io-components/src/components/io-checkbox-group/io-checkbox-group.lifecycle.spec.ts [hunk 2]
```diff
@@ -99,3 +115,43 @@ describe('io-checkbox-group — render() branch coverage', () => {
...
+    (c as any).componentWillLoad();
+    const cb1 = Object.assign(document.createElement('io-checkbox'), { name: '', disabled: false, required: false, state: 'none' });
+    const cb2 = Object.assign(document.createElement('io-checkbox'), { name: '', disabled: false, required: false, state: 'none' });
+    host.appendChild(cb1);
+    host.appendChild(cb2);
+    c.required = true;
+    (c as any).syncChildren();
+    expect((cb1 as any).required).toBe(true);
+    expect((cb2 as any).required).toBe(true);
+  });
```
[the remark this answers is not in the document]
Comment (jakeortega-io) [reply io-components/src/components/io-checkbox-group/io-checkbox-group.lifecycle.spec.ts:131]: Fixed in commit dab3303e. Tests that asserted 'required' was propagated to children have been removed from io-checkbox-group.lifecycle.spec.ts. The spec now only tests name, disabled, and error propagation.

File: io-components/src/components/io-checkbox/io-checkbox.tsx [hunk 3]
```diff
@@ -172,6 +172,7 @@ export class IoCheckbox {
 
   private syncFormValue() {
     // Unchecked checkbox: null = excluded from FormData (matches native checkbox behaviour)
+    // indeterminate does not count as checked — required + indeterminate is still valueMissing
     this.internals?.setFormValue?.(this.checked ? this.value : null);
```
[the remark this answers is not in the document]
Comment (jakeortega-io) [reply io-components/src/components/io-checkbox/io-checkbox.tsx:176]: Fixed in commit dab3303e. Narrowed the comment to be more precise: 'Indeterminate state does not affect form value or validity — only checked/unchecked matters.' This accurately reflects what syncFormValue() actually does.
